clientv3/integration/watch_test.go

```diff
@@ -396,3 +396,80 @@ func TestWatchCompactRevision(t *testing.T) {
 		t.Fatalf("expected closed channel, but got %v", wresp)
 	}
 }
+
+func TestWatchWithProgressNotify(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	tests := []struct {
+		currentRevision int64
+		events          []*storagepb.Event
+		compactRevision int64
+		canceled        bool
+	}{
+		{1, []*storagepb.Event{}, 0, false},
+		{1, []*storagepb.Event{}, 0, false},
+		{1, []*storagepb.Event{}, 0, false},
+
+		{2, []*storagepb.Event{{
+			Type: storagepb.PUT,
+			Kv:   &storagepb.KeyValue{Key: []byte("key_0"), Value: []byte("value"), CreateRevision: 2, ModRevision: 2, Version: 1},
+		}}, 0, false},
+
+		{3, []*storagepb.Event{{
+			Type: storagepb.PUT,
+			Kv:   &storagepb.KeyValue{Key: []byte("key_1"), Value: []byte("value"), CreateRevision: 3, ModRevision: 3, Version: 1},
+		}}, 0, false},
+
+		{3, []*storagepb.Event{}, 0, false},
+		{3, []*storagepb.Event{}, 0, false},
+	}
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 3})
+	defer clus.Terminate(t)
+
+	ready := make(chan struct{})
+	go func() {
+		<-ready
+		kvc := clientv3.NewKV(clus.RandClient())
+		for i := 0; i < 2; i++ {
+			if _, err := kvc.Put(context.TODO(), fmt.Sprintf("key_%d", i), "value"); err != nil {
+				t.Fatal(err)
+			}
+		}
+	}()
+
+	w := clientv3.NewWatcher(clus.RandClient())
+	defer w.Close()
+
+	reportInterval := time.Second
+	v3rpc.ProgressReportInterval = reportInterval
+	rch := w.Watch(context.Background(), "key", clientv3.WithPrefix(), clientv3.WithProgressNotify())
```

Comment:
I think there needs to be a w.Watch(ctx, "key", clientv3.WithProgressNotify()) case too; the watcher doesn't receive any PUT events but it still needs to be notified that the revision is increasing.